### ISSUE TITLE
Consider utilizing a SASS map for variables

### ISSUE THREAD

**Issue by magnificode** [JDRF/design-system#1]
Since we have a very very hefty variables partial, it may be beneficial organization-wise to utilize SASS maps in some areas, with an accompanying function to pluck values from that map (See http://codepen.io/jakealbaugh/post/using-sass-functions-to-access-complex-variable-maps for a great explanation on how to accomplish this.)

In areas like https://github.com/JDRF/design-system/blob/master/src/scss/_variables.scss#L71 and pretty much any other variable that is prefixed with $font- we can employ a SASS map such as the following:
```scss
$font: (
	weight: (
		thin: "Gotham Thin", sans-serif,
		light: "Gotham Light", sans-serif,
		light-italic: "Gotham Light Italic", sans-serif,
		regular: "Gotham Book", sans-serif,
		italic: "Gotham Book Italic", sans-serif,
		medium: "Gotham Medium", sans-serif,
		medium-italic: "Gotham Medium Italic", sans-serif
	),
	size: (
		size-lg: 0.875rem, /* 14px */
		size-sm: 0.8125rem, /* 13px */
		size-h1: 2rem, /* 32px */
		size-h1-sm: 2.25rem, /* 36px */
		size-h1-md: 3rem, /* 48px */
		size-h2: 1.5rem, /* 24px */
		size-h2-sm: 1.75rem, /* 28px */
		size-h2-md: 2.25rem, /* 36px */
		size-h3: 1.25rem, /* 20px */
		size-h3-sm: 1.5rem, /* 24px */
		size-h4: 1.25rem,
		size-h5: 1rem,
		size-h6: 0.875rem
	)
);
```

This approach will hopefully help to keep our variables partial nice and tidy!

**Comment by RachelRVasquez** [comment]
@magnificode 

I gave this a try locally and realized some of the obstacles we'll have if we went down this route.

The sass map variables work only for files where we're both overwriting the variables as well as the styles - as in they're disconnected from their Bootstrap imports. For components we haven't disconnected or variables we aren't overwriting, in order to pursue this path, we'd have to not only change all variables we're overwriting from Bootstrap, but disconnect all components from their imports. This way we can control where those variables are called and replace it with get_map.

Therefore due to the complexity and possibility that we may have to disconnect a lot of styles even if we're not overwriting, for the sake of maintainability, it makes sense not to use sass map vars for the DS at this point in time.
